src/lib.rs

```diff
@@ -78,6 +98,12 @@ pub trait OpenatDirExt {
     /// Remove all content at the target path, returns `true` if something existed there.
     fn remove_all<P: openat::AsPath>(&self, p: P) -> io::Result<bool>;
 
+    /// Update timestamps (both access and modification) to the current time.
+    ///
+    /// If the entry at `path` is a symlink, its direct timestamps are updated without
+    /// following the link.
```

Comment:
Minor: feels like we could reduce this to the usual "Symlinks are not followed."

Reply:
This does actually follow **intermediate** symlinks, and there are consumers relying on that behavior.
For example, the code updating `/ostree/repo` timestamps relies on the intermediate `/ostree -> sysroot/ostree` symlink to reach the actual `/sysroot/ostree/repo` entry.

I guess that's how all the other "not following symlinks" logic works, but I had a doubt today and did double-check it. This docstring reflects that.

Reply:
Right yeah, I agree that's the usual logic (because the alternative would just be erroring out if there's a symlink anywhere in the middle), but agreed it's a good call out.